osis_document/contrib/post_processing/merger.py

```diff
@@ -0,0 +1,86 @@
+# ##############################################################################
+#
+#    OSIS stands for Open Student Information System. It's an application
+#    designed to manage the core business of higher education institutions,
+#    such as universities, faculties, institutes and professional schools.
+#    The core business involves the administration of students, teachers,
+#    courses, programs and so on.
+#
+#    Copyright (C) 2015-2023 Université catholique de Louvain (http://www.uclouvain.be)
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU General Public License as published by
+#    the Free Software Foundation, either version 3 of the License, or
+#    (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU General Public License for more details.
+#
+#    A copy of this license - GNU General Public License - is available
+#    at the root of the source code of this program.  If not,
+#    see http://www.gnu.org/licenses/.
+#
+# ##############################################################################
+import uuid
+from pathlib import Path
+from uuid import UUID
+
+from django.core.files import File
+from pypdf import PdfMerger
+
+from backoffice.settings.base import OSIS_UPLOAD_FOLDER
+from osis_document.enums import PostProcessingType
+from osis_document.exceptions import FormatInvalidException
+from osis_document.models import Upload, PostProcessing
+from osis_document.utils import calculate_hash
+
+
+class Merger:
+    def process(self, input_uuid_files: list, filename=None) -> UUID:
+        input_files = Upload.objects.filter(uuid__in=input_uuid_files)
+        if not input_files:
+            input_files = Upload.objects.filter(output_files__uuid__in=input_uuid_files)
```

Comment:
Pour éviter deux potentielles requêtes et vérifier qu'il n'y a pas de fichier manquant

```suggestion
        input_files = Upload.objects.filter(
            Q(uuid__in=input_uuid_files)
            | Q(output_files__uuid__in=input_uuid_files)
        ).distinct('uuid')
        if len(input_files) != len(input_uuid_files):
            raise MissingFileException
```